include/Api.hpp

```diff
@@ -0,0 +1,301 @@
+    API(std::string path, int port) {
+        db = std::make_shared<Database>(path);
+
+        CROW_ROUTE(app, "/user/get/<int>")
+        ([&](int id) {
+            crow::json::wvalue x({});
+            auto user = db->users->get(id);
+            auto userdata = db->userdata->get(db->userdata->get_where("userId", std::format("{}",id)).front());
+            if (id < 0){
+                x["error"] = "Invalid id";
+                return x;
+            }
+            x["username"] = user["username"];
+            x["password"] = user["password"];
+            x["id"] = user["id"];
+            x["userdataId"] = user["userdataId"];
+            x["stress level"] = 100;
+            x["agegroup"] = userdata["agegroup"];
+            x["occupation"] = userdata["occupation"];
+            return x;
+        });
+        CROW_ROUTE(app, "/user/ids/<int>-<int>")
+        ([&](int min, int max) {
+            crow::json::wvalue x;
+            if (min < 0 || max < 0 || (max-min < 0)){
+                x["difference"] = max-min;
+                x["max"] = max;
+                x["min"] = min;
+                x["error"] = "Invalid Range";
+                return x;
+            }
+            auto users = db->users->get_where();
+            std::vector<int> filteredUsers;
+            for(const auto & user : users) {
+                if(user <= max && user >= min) {
+                    filteredUsers.push_back(user);
+                }
+            }
+            x = std::move(filteredUsers); //Funky solution to make it only return json array
+            return std::move(x);          //This is done due to "inconsistent type" errors appearing if not done
+        });
+        CROW_ROUTE(app, "/user/ids")
+        ([&]() {
+            crow::json::wvalue x;
+            auto users = db->users->get_where();
+            x = std::move(users);
+            return std::move(x);
+        });
+        
+        CROW_ROUTE(app, "/auth")
+        .methods("POST"_method)
+        ([&](const crow::request& req) {
+            auto x = crow::json::load(req.body);
+            //crow::json::wvalue responsejson({});
+            if (!x){
+                return crow::response(400, "Unable to load/parse json");
+            }
+        
+            auto username = x["username"].s();
+            auto password = x["password"].s();
+            auto dbpassword = db->users->get(db->users->get_where("username", username).front())["password"];
+            if(dbpassword == password){
+                auto userid = db->users->get_where("username", username).front();
+                auto hash = Hash((std::string)username, (std::string)password);
+                auto token = std::format("{}", hash);
+                authedUsers[userid] = token;
+                return crow::response(200, token);
+            }
+            else{
+                return crow::response(403, "Invalid Token");
+            }
+        });
+        CROW_ROUTE(app, "/register")
```

Comment:
shouldn't auth and register be sorted under /user like: `/user/auth` and `/user/register`

Reply:
done